src/auth-service/controllers/join.js

```diff
@@ -283,73 +354,90 @@ const join = {
         });
       }
     } catch (e) {
-      return res.status(HTTPStatus.BAD_REQUEST).json({
+      return res.status(HTTPStatus.BAD_GATEWAY).json({
         success: false,
-        message:
-          "please crosscheck the api query parameters using the documentation",
+        message: "there is a server error",
         error: e.message,
       });
     }
   },
 
   getDefaults: async (req, res) => {
     try {
-      const { user, chartTitle } = req.query;
-      let filter = {};
-      console.log;
-      if (!isEmpty(user) && isEmpty(chartTitle)) {
-        filter = {
-          user: user,
-        };
-      } else if (!isEmpty(user) && !isEmpty(chartTitle)) {
-        filter = {
-          user: user,
-          chartTitle: chartTitle,
-        };
+      //....
+      logText(".....................................");
+      logText("list all defaults by tenant...");
+      const { tenant, user, chartTitle } = req.query;
+
+      if (tenant && user && !chartTitle) {
+        logElement("the tenant", tenant);
+        logElement("the user", user);
+        const defaults = await DefaultModel(tenant).find({ user: user }).exec();
+        logObject("the defaults", defaults);
+        if (!isEmpty(defaults)) {
+          return res.status(HTTPStatus.OK).json({
+            success: true,
+            message: `Customised chart defaults for ${user} fetched successfully`,
+            defaults,
+          });
+        } else if (isEmpty(defaults)) {
+          return res.json({
+            success: false,
+            message: `this organisation (${tenant}) does not have these user defaults or they do not exist, please crosscheck`,
```

Comment:
@Baalmart I think it's better to return an empty array for users with no defaults set instead for an error.

Reply:
I think this was addressed yesterday. Please test and share feedback. Thanks @ja-odur